lib/data_hygiene/publishing_api_sync_check.rb

```diff
@@ -0,0 +1,113 @@
+# Check to see how complete the (live) content-store's copies of data are.
+#
+#   pasc = DataHygiene::PublishingApiSyncCheck.new(StatisticsAnnouncement.limit(100))
+#   pasc.add_expectation { |json, model| json.fetch("format") == "statistics_announcement" }
+#   pasc.perform
+#
+# The disadvantages of running against content-store are that you can't inspect
+# things like rendering_app.
+#
+# On the other hand, if you don't trust publishing-api's ability to push to
+# content-store...
+#
+# If we used publishing-api, we could use it's index endpoint to do this all in
+# a single request. The problem is that that endpoint currently doesn't do
+# pagination, so it won't (reliably) handle lots of items...
+module DataHygiene
+  class PublishingApiSyncCheck
+    class Success
+      attr_reader :base_path
+
+      def initialize(base_path:)
+        @base_path = base_path
+      end
+    end
+
+    class Failure
+      attr_reader :base_path, :failed_expectations
+
+      def initialize(options)
+        @base_path = options[:base_path]
+        @failed_expectations = options[:failed_expectations]
+      end
+
+      def to_s
+        "Failed path: #{@base_path}, failed expectations: #{@failed_expectations.join(', ')}"
+      end
+
+      def ==(other)
+        self.base_path == other.base_path && self.failed_expectations == other.failed_expectations
+      end
+    end
+
+    attr_reader :hydra, :scope, :expectations, :successes, :failures, :base_path_builder
+
+    def initialize(scope)
+      @scope = scope
+      Ethon.logger = Logger.new(nil) # disable Typhoeus/Ethon debug logging
+      @hydra = Typhoeus::Hydra.new(max_concurrency: 20)
```

Comment:
I wonder if the concurrency should be lower in order to reduce the number of in-flight requests, since the limit per server/proxy is probably much lower also.